iib/workers/tasks/build_merge_index_image.py

```diff
@@ -153,31 +155,31 @@ def _add_bundles_missing_in_source(
     )
     _add_label_to_index(
         'com.redhat.index.delivery.distribution_scope',
-        distribution_scope,
+        distribution_scope,  # type: ignore
```

Comment:
May I ask in what situations you have used type: ignore.
I noticed its used in function calls, local variables and some others

Reply:
Yes, of course. We discussed that with @yashvardhannanavati . We did not wanted to do big changes to logic of the IIB with this PR. 
Adding the types resulted in Mypy errors, which few were fixed, however few of them cannot be fixed without changing of the IIB logic (most of the times checking if the parameter is not None, I think this example is that case too).
So places where those errors were, are now annotated with those `# type: ignore` comments. After this PR is merged I will create another PR, where those comments will be removed, and fixes which are removing those errors proposed. 
Or I can make another commit which will fix those mypy errors and add it to this PR.

Reply:
Another PR would be better :+1: